packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -1311,4 +1326,283 @@ export class KubernetesClient {
     }
   }
+
+  async restartPod(name: string): Promise<void> {
+    let telemetryOptions = {};
+    try {
+      const ns = this.currentNamespace;
+      const connected = await this.checkConnection();
+      if (!ns) {
+        throw new Error('no active namespace');
+      }
+      if (!connected) {
+        throw new Error('not active connection');
+      }
+
+      const pod = await this.readNamespacedPod(name, ns);
+      if (!pod?.metadata) {
+        throw new Error('no metadata found');
+      }
+
+      const creationSource = this.checkPodCreationSource(pod.metadata);
+      if (creationSource.isManuallyCreated) {
+        await this.restartManuallyCreatedPod(name, ns, pod);
+      } else {
+        if (!creationSource.controllerType) {
+          throw new Error('unable to restart controlled pod');
+        }
+
+        const controller = this.getPodController(pod.metadata);
+        const controllerName = controller!.name;
+
+        if (isScalableControllerType(creationSource.controllerType)) {
+          await this.scaleControllerToRestartPods(ns, controllerName, creationSource.controllerType);
+        } else if (creationSource.controllerType === 'Job') {
+          await this.restartJob(controllerName, ns);
+        }
+      }
+    } catch (error) {
+      telemetryOptions = { error: error };
+      throw this.wrapK8sClientError(error);
+    } finally {
+      this.telemetry.track('kubernetesRestartPod', telemetryOptions);
+    }
+  }
+
+  protected async restartManuallyCreatedPod(name: string, namespace: string, pod: V1Pod): Promise<void> {
+    const coreApi = this.kubeConfig.makeApiClient(CoreV1Api);
+    await coreApi.deleteNamespacedPod(name, namespace);
+
+    const isDeleted = await this.waitForPodDeletion(coreApi, name, namespace);
+    if (!isDeleted) {
+      throw new Error(`pod "${name}" in namespace "${namespace}" was not deleted within the expected timeframe`);
+    }
+
+    delete pod.metadata?.resourceVersion;
+    delete pod.metadata?.uid;
+    delete pod.metadata?.selfLink;
+    delete pod.metadata?.creationTimestamp;
+    delete pod.status;
+
+    const newPod: V1Pod = { ...pod };
+    await coreApi.createNamespacedPod(namespace, newPod);
+  }
+
+  protected async waitForPodDeletion(
+    coreApi: CoreV1Api,
+    name: string,
+    namespace: string,
+    timeout: number = 60000,
+  ): Promise<boolean> {
+    const startTime = Date.now();
+
+    while (Date.now() - startTime < timeout) {
+      try {
+        await coreApi.readNamespacedPodStatus(name, namespace);
+        await new Promise(resolve => setTimeout(resolve, 1000));
+      } catch (e) {
+        const error = e ?? {};
+        if (typeof error === 'object' && 'response' in error) {
+          const axiosError = error as { response: { statusCode: number } };
+          if (axiosError.response.statusCode === 404) {
+            return true;
+          }
+        }
+        throw e;
```

Comment:
nit: you are correct, but still can be shorter and without additional constant. It is up to you just to resolve this conversation without changes.

```suggestion
        if (e && typeof e === 'object' && 'response' in e) {
          const axiosError = e as { response: { statusCode: number } };
          if (axiosError.response.statusCode === 404) {
            return true;
          }
        }
        throw e;
```